### PR TITLE
build-bottle-pr: Add option --keep-old

### PR DESCRIPTION
**Pull request by sjackman**

### CODE REVIEW

File: cmd/brew-build-bottle-pr.rb [hunk 1]
```diff
@@ -108,6 +116,7 @@ def build_bottle(formula)
       safe_system "git", "checkout", "-b", branch, "master"
       unless ARGV.dry_run?
         safe_system "git", "commit", formula.path, "-m", message
+        keep_old if ARGV.include? "--keep-old"
```
Comment (rwhogg) [inline cmd/brew-build-bottle-pr.rb]: Should this be added to the help text at the top?

File: cmd/brew-build-bottle-pr.rb [hunk 2]
```diff
@@ -77,6 +77,14 @@ def check_remotes(formulae)
     end
   end
 
+  def keep_old
```
Comment (maxim-belkin) [inline cmd/brew-build-bottle-pr.rb]: would be good to add a description above the formula ... OR...  let's, actually, start documenting our functions!

File: cmd/brew-squash-bottle-pr.rb [hunk 3]
```diff
@@ -23,6 +28,12 @@ def squash_bottle_pr
     ENV["GIT_EDITOR"] = git_editor
 
     safe_system "git", "show" if ARGV.verbose?
+
+    if Utils.popen_read("git", "log", "-n1", "--pretty=%s", "HEAD~1") =~ /^drop! /
+      bottle_head = Utils.popen_read("git", "rev-parse", "HEAD").chomp
+      safe_system "git", "reset", "--hard", "HEAD~2"
+      safe_system "git", "cherry-pick", bottle_head
+    end
```
Comment (maxim-belkin) [inline cmd/brew-squash-bottle-pr.rb]: damn! that's tricky :)

File: cmd/brew-squash-bottle-pr.rb [hunk 4]
```diff
@@ -7,6 +7,11 @@ module Homebrew
   #    brew pull --bottle 123
   #    brew squash-bottle-pr
   def squash_bottle_pr
+    unless Utils.popen_read("git", "log", "-n1", "--pretty=%s", "HEAD~1") =~ /: Build a bottle for Linuxbrew$/
+      opoo "No build-bottle-pr commit was found"
```
Comment (maxim-belkin) [inline cmd/brew-squash-bottle-pr.rb]: `opoo` or `ohai`?

**Comment by maxim-belkin** [comment]
LGTM!
My only suggestion would be to think about `keep-mac-bottles` option. Lengthy name but easy to understand. This way we can pre-process formulas by removing `x86_64_linux` bottle.

File: cmd/brew-squash-bottle-pr.rb (hunk 3 above)
Comment (sjackman) [inline cmd/brew-squash-bottle-pr.rb]: 😈

File: cmd/brew-build-bottle-pr.rb (hunk 2 above)
Comment (sjackman) [inline cmd/brew-build-bottle-pr.rb]: Hah! That would be something! =)
I've added `# Add --keep-old to circle.yml`

File: cmd/brew-squash-bottle-pr.rb (hunk 4 above)
Comment (sjackman) [inline cmd/brew-squash-bottle-pr.rb]: I think a warning, if `squash_bottle_pr` is called they're should be a `Build a bottle for Linuxbrew` commit, and give a warning if not.

**Comment by sjackman** [comment]
Merged.

**Pull request closed without merging**